internal/ask/ask.go

```diff
@@ -0,0 +1,205 @@
+            Message: components.Message{
+                AssistantMessage: &components.AssistantMessage{
+                    Content: flags.Message,
+                },
+            },
+		})
+		if err != nil {
+            SessionID = res.ChatResult.ChatSessionID
+			handleError(logger, err)
+			return SessionID, err
+		}
+
+        if res.ChatResult != nil {
+            printWithFootnotes(ctx, res.ChatResult.Message.Content)
+        } else {
+            logger.Error("\nNo chat response received.")
+        }
+	}
+
+	return SessionID, nil 
+}
+
+
+func handleError(logger log.Logger, err error) {
+	switch e := err.(type) {
+	case *sdkerrors.HTTPValidationError:
+		logger.Errorf("HTTP Validation Error: %v", e)
+	case *sdkerrors.SDKError:
+		logger.Errorf("SDK Error: %v", e)
+	default:
+		logger.Errorf("Error: %v", err)
+	}
+}
+
+func processMarkdown(text string) string {
+    text = boldRegex.ReplaceAllStringFunc(text, func(match string) string {
+        return strings.ToUpper(match[2 : len(match)-2])
+    })
+
+    text = italicRegex.ReplaceAllStringFunc(text, func(match string) string {
+        return match[1 : len(match)-1] 
+    })
+
+    return text
+}
+
+
+func printWithFootnotes(ctx context.Context, text string) {
+	logger := log.From(ctx)
+    text = processMarkdown(text)
+
+    // Transform footnotes
+    matches := linkRegex.FindAllStringSubmatch(text, -1)
+
+    if len(matches) == 0 {
+        logger.Printf(text)
+        return
+    }
+
+    footnotes := make(map[string]string)
+    var orderedRefs []string
+
+    for _, match := range matches {
+        refName, url := match[1], match[2]
+        if _, exists := footnotes[refName]; !exists {
+            orderedRefs = append(orderedRefs, refName)
+        }
+        footnotes[refName] = url
+        // Directly replace the markdown link with the reference name, removing parentheses in the process
+        text = strings.Replace(text, match[0], "["+refName+"]", 1)
+    }
+
+    logger.Printf("\n%s",text)
+    logger.PrintfStyled(styles.Focused, "\nReferences:")
+    for _, ref := range orderedRefs {
+        logger.PrintfStyled(styles.Dimmed, "[%s]: %s\n", ref, footnotes[ref])
+    }
+}
+
+
+
+func RunInteractiveChatSession(ctx context.Context, initialFlags AskFlags) error {
+    logger := log.From(ctx)
+	sessionID := "" 
+	scanner := bufio.NewScanner(os.Stdin)
+    logger.Info("Entering interactive chat session, type exit to quit.")
+
+	if initialFlags.Message != "" {
+		logger.Info("\nProcessing your question...")
+		var err error
+		sessionID, err = Ask(ctx, initialFlags)
+		if err != nil {
+            logger.Errorf("An error occurred while processing question, ending chat: %v", err)
+			return err
+		}
+	}
+
+	for {
+        promptStyle := styles.Focused.Render("> ") 
+        fmt.Print(promptStyle) 
+		if !scanner.Scan() {
+			break
+		}
+
+		input := scanner.Text()
+		if input == "exit" {
+            logger.Info("Exiting chat session.")
```

Comment:
Seems like some of the formatting is off here. I think `gofmt -w .` from the root will do the trick